### PR TITLE
Feature: Cart

### PR DESCRIPTION
**Pull request by ksookdhew**
Screenshot 2024-08-08 at 09 33 27

### CODE REVIEW

File: src/features/navbar/Navbar.tsx [hunk 1]
```diff
@@ -22,20 +25,32 @@ const Navbar = () => {
                         </label>
                     </div>
                     <div className="mx-2 flex-1 px-2 text-xl">SHOP</div>
-                    <div className="hidden flex-none lg:block">
+                    <div className="hidden flex-none lg:block px-8">
                         <ul className="menu menu-horizontal text-xl">
-                            <li><a>Wishlist</a></li>
-                            <li><a>Cart</a></li>
+                            <li><Link to='/'>Products</Link></li>
+                            <li><a>Contact Us</a></li>
```
Comment (MarcusNkadimeng) [inline src/features/navbar/Navbar.tsx:31]: In future, don't leave things that are not working or have no use yet. :)

**Review by MarcusNkadimeng — APPROVED** [review]

File: src/features/cart/cartStore.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return state;
+    }),
+    removeFromCart: (id: number) => set((state) => {
+        const newCart = state.cart.filter(item => item.productId !== id);
+        return {
+            cart: newCart,
+            cartNumItems: calculateTotalItems(newCart),
+        };
+    }),
+    clearCart: () => set(() => ({
```
Comment (AyaazStreet) [inline src/features/cart/cartStore.ts:73]: Unused function

**Review by AyaazStreet — CHANGES_REQUESTED** [review]

File: src/features/cart/CartProductCard.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import {useQuery} from "@tanstack/react-query";
+import {getProductWithId} from "../../services/api.ts";
+import {formattedPrice} from "../../utils/productUtils.ts";
+
+const CartProductCard = ({cartItem}: { cartItem: CartItem }) => {
+    const productQuery = useQuery({
+        queryKey: [cartItem.productId],
+        queryFn: () => getProductWithId(`${cartItem.productId}` ?? '')
+    })
+    const addItemToCart = useCartStore((state) => state.addItem);
```
Comment (sjacobs1) [inline src/features/cart/CartProductCard.tsx:11]: consider using derived state for the cart products, so that you don't make calls for each product

**Review by sjacobs1 — APPROVED** [review]

File: src/features/cart/cartStore.ts (hunk 2 above)
Comment (ksookdhew) [reply src/features/cart/cartStore.ts:73]: Good spot! Thanks 😄

File: src/features/cart/CartProductCard.tsx (hunk 3 above)
Comment (ksookdhew) [inline src/features/cart/CartProductCard.tsx:11]: I think there is a little confusion here with derived state 😄

**Review by AyaazStreet — APPROVED** [review]
🤙🏻

**Pull request merged**